### PR TITLE
Reference Object and Schema Object use of $ref updates for 2019-09 / OAS 3.1

### LINKED ISSUE #1887: $ref value must be accessible URL for external reference?
Based on JSON reference spec, the $ref value is an URI, which by definition is a resource identifier only, and is independent of access.
It seems it is valid to use a well formed URI, but not an accessible URL (assuming http).
However, if it is not a valid URL for external reference, how would different tools resolve the references?

### PR DESCRIPTION
**Pull request by handrews**
Partially handles #2092 
Fixes #1887

### CODE REVIEW

**Comment by webron** [comment]
\*ahem\* https://github.blog/2019-02-14-introducing-draft-pull-requests/

**Comment by handrews** [comment]
😮 
Since no one has commented on this anyway I'll close it and open one of those- I had no idea draft PRs were a thing!

**Comment by handrews** [comment]
Except this repo doesn't seem to offer the draft option?  I'm confused so I'm re-opening this- I just went through all of the ways to set up a PR I could think of and none of them gave me that drop down on the button.

File: versions/3.1.0.md [hunk 1]
```diff
@@ -2290,7 +2288,7 @@ $ref: definitions.yaml#/Pet
 
 #### <a name="schemaObject"></a>Schema Object
 
-The Schema Object allows the definition of input and output data types. These types can be objects, but also primitives and arrays. This object is a superset of the [JSON Schema Draft 2019-09](http://json-schema.org/) specification.
+The Schema Object allows the definition of input and output data types. These types can be objects, but also primitives and arrays. This object is a superset of the [JSON Schema Specification Draft 2019-09](http://json-schema.org/) specification.
```
Comment (MikeRalphson) [inline versions/3.1.0.md]: Can we lose the final "specification"?

Comment (philsturgeon) [reply versions/3.1.0.md]: I think it's correct the way it is. We don't shove Specification into the name of our project we just accept that its a project which happens to have a specification. :)

Comment (MikeRalphson) [reply versions/3.1.0.md]: Do you mean this is ok

> This object is a superset of the JSON Schema **Specification** Draft 2019-09 **specification**.

Or that the link should not be changed?

Comment (handrews) [reply versions/3.1.0.md]: @MikeRalphson @philsturgeon I changed it to make it consistent with all of the other references in the OAS spec.  I don't care whether the word is there or not, but it should either always be there or never be there.

File: versions/3.1.0.md [hunk 2]
```diff
@@ -170,12 +170,12 @@ The formats defined by the OAS are:
 Throughout the specification `description` fields are noted as supporting CommonMark markdown formatting.
 Where OpenAPI tooling renders rich text it MUST support, at a minimum, markdown syntax as described by [CommonMark 0.27](https://spec.commonmark.org/0.27/). Tooling MAY choose to ignore some CommonMark features to address security concerns. 
 
-### <a name="relativeReferences"></a>Relative References in URLs
+### <a name="relativeReferences"></a>Relative References in URIs
 
-Unless specified otherwise, all properties that are URLs MAY be relative references as defined by [RFC3986](https://tools.ietf.org/html/rfc3986#section-4.2).
-Relative references are resolved using the URLs defined in the [`Server Object`](#serverObject) as a Base URI.
+Unless specified otherwise, all properties that are URIs MAY be relative references as defined by [RFC3986](https://tools.ietf.org/html/rfc3986#section-4.2).
+Relative references are resolved using the URLs defined in the [`Server Object`](#serverObject) as a Base URI.  This includes relative references in [`Reference Objects`](#referenceObjects), outside of any [`Schema Object`](#schemaObject)l
```
Comment (darrelmiller) [inline versions/3.1.0.md]:
```suggestion
Relative references are resolved using the URLs defined in the [`Server Object`](#serverObject) as a Base URI.  This includes relative references in [`Reference Objects`](#referenceObjects), outside of any [`Schema Object`](#schemaObject).
```

File: versions/3.1.0.md [hunk 3]
```diff
@@ -170,12 +170,12 @@ The formats defined by the OAS are:
 Throughout the specification `description` fields are noted as supporting CommonMark markdown formatting.
 Where OpenAPI tooling renders rich text it MUST support, at a minimum, markdown syntax as described by [CommonMark 0.27](https://spec.commonmark.org/0.27/). Tooling MAY choose to ignore some CommonMark features to address security concerns. 
 
-### <a name="relativeReferences"></a>Relative References in URLs
+### <a name="relativeReferences"></a>Relative References in URIs
 
-Unless specified otherwise, all properties that are URLs MAY be relative references as defined by [RFC3986](https://tools.ietf.org/html/rfc3986#section-4.2).
-Relative references are resolved using the URLs defined in the [`Server Object`](#serverObject) as a Base URI.
+Unless specified otherwise, all properties that are URIs MAY be relative references as defined by [RFC3986](https://tools.ietf.org/html/rfc3986#section-4.2).
+Relative references are resolved using the URLs defined in the [`Server Object`](#serverObject) as a Base URI.  This includes relative references in [`Reference Objects`](#referenceObjects), outside of any [`Schema Object`](#schemaObject).
```
Comment (MikeRalphson) [inline versions/3.1.0.md]: The build failure here is related to the `#referenceObjects` plural, which I think should be singular.
```suggestion
Relative references are resolved using the URLs defined in the [`Server Object`](#serverObject) as a Base URI.  This includes relative references in [`Reference Objects`](#referenceObject), outside of any [`Schema Object`](#schemaObject).
```

Comment (MikeRalphson) [reply versions/3.1.0.md]: @handrews suggestion for ya.

File: versions/3.1.0.md (hunk 1 above)
Comment (philsturgeon) [reply versions/3.1.0.md]: I misread the diff @MikeRalphson, what you said is correct. 👇 


```suggestion
The Schema Object allows the definition of input and output data types. These types can be objects, but also primitives and arrays. This object is a superset of the [JSON Schema Specification Draft 2019-09](http://json-schema.org/).
```

**Pull request merged**